### PR TITLE
Modified binary search

### LINKED ISSUE #1224: Updating files to take input at runtime (#1160)
Updating these files : 
Binary_Search.cpp
Binary_Search.java
Binary_Search.go
Binary_Search.py
searching_a_circularly_sorted_array.cpp

### PR DESCRIPTION
**Pull request by singh-shreya6**
Fixes #1224 referring #1160 

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [x] Input is taken dynamically.
- [x] Sample Input / Output is added at the end of file.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

#### Changes proposed in this pull request:
- Modified Binary search files for runtime input

#### Languages Used:
- Python
- C++
- Java
- Go

#### Files Added:
- Binary_Search.cpp 
- Binary_Search.py
- Binary_Search.java
- Binary_Search.go
- searching_a_circularly_sorted_array.cpp

### CODE REVIEW

**Comment by GOVINDDIXIT** [comment]
@singh-shreya6 You have to mention the parent issue in the PR description

**Comment by singh-shreya6** [comment]
Done.

File: Binary_Search/Binary_Search.cpp [hunk 1]
```diff
@@ -62,42 +62,44 @@ int Upper_Bound(int array[], int size, int desired)
...
-    if (Binary_Search(array, 7, 9) != -1)
+    int num;
+    scanf("%d", &num);
+    int array[num];
+    for (int i = 0; i < num; i++) {
+        scanf("%d", &array[i]);
+    }
+    int desired;
+    scanf("%d", &desired);
+    if (Binary_Search(array, num, desired) != -1)
```
Comment (sridharjajoo) [inline Binary_Search/Binary_Search.cpp]: please change the function name to binarySearch. The above kind of naming is generally followed in c#

**Comment by singh-shreya6** [comment]
@jainaman224 I don't know why it shows closed, while it got merged.

**Comment by jainaman224** [comment]
You have forced pushed it in incorrect way.

**Pull request closed without merging**